base/sparse/cholmod.jl

```diff
@@ -87,22 +87,22 @@ function __init__()
             tmp = Array(Cint, 3)
             ccall((:cholmod_version, :libcholmod), Cint, (Ptr{Cint},), version_array)
             ccall((:jl_cholmod_version, :libsuitesparse_wrapper), Cint, (Ptr{Cint},), tmp)
-            if tmp != version_array
+            if tmp[1] != version_array[1]
```

Comment:
I'm not sure that the versioning policy in SuiteSaprse has been strict enough that testing only the major version is enough (though I initially thought that). E.g. the version number function for CHOLMOD was introduced in 2.1.0. I briefly looked at the changelog where I saw that
```
so although it is a major version number change for CHOLMOD it is not for the other libraries even though the API for providing custom memory allocation function has changed.


March 26, 2014: SuiteSparse version 4.3.0
    MAJOR UPDATE: first release of CHOLMOD GPU acceleration
    minor update: modified all packages to use SuiteSparse_config for
        malloc/calloc/realloc/free, printf, hypot, and divcomplex
        function pointers.

    * AMD 2.4.0:
        changed malloc/printf pointers to use SuiteSparse_config
    * CAMD 2.4.0:
        changed malloc/printf pointers to use SuiteSparse_config
    * CCOLAMD 2.9.0:
        changed malloc/printf pointers to use SuiteSparse_config
    * CHOLMOD 3.0.0:
        MAJOR UPDATE to GPU acceleration.  Released at GTC 2014.
    * COLAMD 2.9.0:
        changed malloc/printf pointers to use SuiteSparse_config
    * CSparse 3.1.3:
        minor update to UFget
    * CXSparse 3.1.3:
        minor update to UFget
    * KLU 1.3.0:
        changed malloc/printf pointers to use SuiteSparse_config
    * LDL 2.2.0:
        changed malloc/printf pointers to use SuiteSparse_config
    * RBio 2.2.0:
        changed malloc/printf pointers to use SuiteSparse_config
    * SPQR 1.3.2:
        changed malloc/printf pointers to use SuiteSparse_config
    * UMFPACK 5.7.0:
        changed malloc/printf pointers to use SuiteSparse_config
    * MATLAB_Tools:
        added stats to UFcollection, revised commends in SFMULT,
        minor update to spok
```